pkg/repository/prisma/dbsqlc/workflow_runs.sql

```diff
@@ -1551,3 +1551,26 @@ WHERE
 DELETE FROM "WorkflowTriggerScheduledRef"
 WHERE
     "id" = @scheduleId::uuid;
+
+-- name: GetUpstreamErrorsForOnFailureStep :many
+WITH workflow_run AS (
+    SELECT wr."id"
+    FROM "WorkflowRun" wr
+    JOIN "JobRun" jr ON wr."id" = jr."workflowRunId"
+    JOIN "StepRun" sr ON jr."id" = sr."jobRunId"
+    WHERE sr."id" = @onFailureStepRunId::uuid
+)
+SELECT
+    sr."id" AS "stepRunId",
+    s."readableId" AS "stepReadableId",
+    sr."error"
+FROM "WorkflowRun" wr
+JOIN "JobRun" jr ON wr."id" = jr."workflowRunId"
+JOIN "StepRun" sr ON jr."id" = sr."jobRunId"
+JOIN "Step" s ON sr."stepId" = s."id"
+JOIN "Job" j ON jr."jobId" = j.id
+WHERE
+    wr."id" = (SELECT "id" FROM workflow_run)
+    -- Don't include the on-failure step in the step runs listed
+    AND j."kind" <> 'ON_FAILURE'
+    AND sr."error" IS NOT NULL;
```

Comment:
would love ideas about how to make this less redundant!

Reply:
You shouldn't need the join on `Step` and `Job` since an onFailureJob is guaranteed to only have a single step run, and only one onFailureJob is defined per WorkflowRun. So something like this would work:

```sql 
SELECT
    sr."id" AS "stepRunId",
    s."readableId" AS "stepReadableId",
    sr."error"
FROM "WorkflowRun" wr
JOIN "JobRun" jr ON wr."id" = jr."workflowRunId"
JOIN "StepRun" sr ON jr."id" = sr."jobRunId"
WHERE 
    sr."stepRunId" != @onFailureStepRunId::uuid
    AND sr."error" IS NOT NULL
```

Reply:
Ah nice, thanks! Do I still need to join to the `Step` to get the `readableId`? or is there a way around that?

Reply:
Ah missed that, yeah there's no way around that

Reply:
Sounds good, will fix the other one 👍